ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionengine/ExecutionPayloadStatus.java

```diff
@@ -14,7 +14,34 @@
 package tech.pegasys.teku.spec.executionengine;
 
 public enum ExecutionPayloadStatus {
-  VALID,
-  INVALID,
-  SYNCING
+  VALID(Validity.VALID),
+  INVALID(Validity.INVALID),
+  SYNCING(Validity.NOT_VALIDATED),
+  ACCEPTED(Validity.NOT_VALIDATED),
+  INVALID_BLOCK_HASH(Validity.INVALID),
+  INVALID_TERMINAL_BLOCK(Validity.INVALID);
```

Comment:
I like this approach.  Am I right in thinking that this would let our internal ForkChoice code handle either the Kintsugi style Engine API or the newer one?  If so, it might be worth considering maintaining support for both versions in the engine service.  Since we know the current version will be deleted reasonably soon we don't have to make it too neat - we could just duplicate `Web3JExecutionEngineClient` to have completely separate old and new versions.  They'd both implement the same interface which would likely use the new `newPayload` method name but the "old" implementation would still send an `engine_executePayload` request for that and it would adapt the responses as required so forkChoiceUpdated `SUCCESS` would become `VALID` etc).

Reply:
it wasn't my primary goal but I see your point. I'll go down to that road and see.

Reply:
@ajsutton this my approach to kintsugi backward compatibility https://github.com/ConsenSys/teku/pull/4923/commits/464c0a3ddb85afb52fa38571291998e7f3c8a35a

Reply:
Yeah that looks good. Would be very nice if we can manage to get this stuff merged without breaking Kintsugi.